ui/components-react/src/components-react/toolbar/Toolbar.tsx

```diff
@@ -48,15 +374,50 @@ export interface ToolbarProps extends CommonProps, NoChildrenProps {
   onItemExecuted?: OnItemExecutedFunc;
   /** Optional function to call on any KeyDown events processed by toolbar */
   onKeyDown?: (e: React.KeyboardEvent) => void;
+  /**
+   * @internal Use [Toolbar]($appui-react) from appui-react instead of using this property.
+   */
+  syncUiEvent?: BeEvent<(args: { eventIds: Set<string> }) => void>;
```

Comment:
Could we avoid adding internal members to public interfaces as those never make sense?

Reply:
How about existing consumers that have NOT moved away from this deprecated component? Do they need to use the internal `syncUiEvent` to correctly handle conditional stuff?

Reply:
We could avoid adding this internal members if we move the whole implementation to appui-react and make a copy of it, which I hope was the plan once we can actually get rid of the components-react implementation, but at this time, I think it's cleaner to add an internal prop. (One other option would be to simply NOT declare it at all and use it anyway, this is only a type and as it is internal, we could cast it so we know its potentially there.)

Existing consumers that have NOT moved away from this deprecated stuff shouldnt expect the new feature, and as such the component will behave like it did before. (which is, no conditionals are handled, which is what happen if you dont provide the event)

Reply:
A third option would be to have a completely internal toolbar implementation in components-react, which the public implementations from component-react and appui-react would use, maybe that's the cleanest.

Reply:
Moved the common implementation to `InternalToolbarComponent` which is now used by `Toolbar`, `ToolbarWithOverflow` in components react (reverted all props changes) and `Toolbar` from appui-react, using the new props. I'll update the enableOverflow in a separate commit but this comment should be resolvable now.